domdiv/main.py

```diff
@@ -812,37 +835,105 @@ def filter_sort_cards(cards, options):
     # Expansion names can be the names from the language or the cardset_tag
     if options.expansions:
-        options.expansions = set([e.lower() for e in options.expansions])
-        wantedExpansions = set()
+        # First get the Official set infomation
+        Official_sets = set()  # Will hold official sets
+        Official_search = []  # Will hold official sets for searching, both set key and set_name
+        for s in Card.sets:
+            if not Card.sets[s].get("fan", False):
+                Official_sets.add(s)
+                Official_search.extend([s.lower(), Card.sets[s].get('set_name', None).lower()])
+
+        # Expand out any wildcards, matching set key or set name in the given language
+        expanded_expansions = []
+        for e in options.expansions:
+            matches = fnmatch.filter(Official_search, e)
+            if matches:
+                expanded_expansions.extend(matches)
+            else:
+                expanded_expansions.append(e)
+
+        # Now get the actual sets that are matched above
+        options.expansions = set([e for e in expanded_expansions])  # Remove duplicates
+        wantedSets = set()
         knownExpansions = set()
-        # Match sets that either start with the expansion set key (used by cardset_tag)
-        # or the actual name of the set/expansion in the specified language.
         for e in options.expansions:
-            for s in Card.sets:
-                if (s.lower().startswith(e) or
-                        Card.sets[s].get('set_name', "").lower().startswith(e)):
-                    wantedExpansions.add(s)
+            for s in Official_sets:
+                if (s.lower() == e or Card.sets[s].get('set_name', "").lower() == e):
+                    wantedSets.add(s)
                     knownExpansions.add(e)
 
         # Give indication if an imput did not match anything
         unknownExpansions = options.expansions - knownExpansions
         if unknownExpansions:
             print "Error - unknown expansion(s): %s" % ", ".join(unknownExpansions)
+            print "Possible options are:\n{}".format('\n'.join(Official_sets))
 
-        # Now keep only the cards that were in the expansions requested
-        filteredCards = []
-        for c in cards:
-            if c.cardset_tag in wantedExpansions:
-                filteredCards.append(c)
-        cards = filteredCards
+        # Now keep only the cards that are in the expansions requested
+        cards = []
+        for c in Official_cards:
+            if c.cardset_tag in wantedSets:
+                cards.append(c)
+
+    # Take care of fan expansions.  Fan expansions must be explicitly named to be added.
+    # If no --fan is given, then no fan cards are added.
+    # Fan expansion names can be the names from the language or the cardset_tag
+    if options.fan:
+        # First get the Fan set infomation
+        Fan_sets = set()  # Will hold fan sets
```

Comment:
Can all of this code be shared with the version for the official expansions? In that case we can add the `none` keyword to both of them and default the fan expansions to use it unless otherwise specified.